data-raw/tidyverse_creation.R

```diff
@@ -0,0 +1,177 @@
+library(dplyr)
+# ------------------------------------------------------------------------------
+body_function <- function(fct, argument, args1) {
+
+  body_arg <- paste0("(", paste(args1, collapse = ", "), ", ...)")
+  body_f <- NULL
+
+  if (length(args1) > 0) {
```

Comment:
This part would be more efficiently written this way (always try to put if-tests outside for-loops as much as you can).
Also, I've replaced the for-loop by functional programming.

```
lgth <- length(args1)
if (lgth > 0) {
  body_f <- paste0(
    body_f,
    '\told_attr <- purrr::keep(attributes(', args1[1],
    '), names(attributes(', args1[1], ')) %in% ',
    'c("dic_r2g", "dic_g2r", "wkdir", "experiment", "model", "class")) \n ',
    '\t', args1[1], ' <- as.data.frame(', args1[1], ') \n')
  if (lght > 1) {
    f <- function(x) {
      paste0(
        '\told_attr <- purrr::keep(attributes(', x,
        '), names(attributes(', x, ')) %in% ',
        'c("dic_r2g", "dic_g2r", "wkdir", "experiment", "model", "class")) \n ',
        '\t', x, ' <- as.data.frame(', x, ') \n')
    }
    body_f <- paste0(c(body_f, lapply(args1[-1], f)), collapse = "")
  }
  body_f
}
```

Reply:
I replace the for-loop by your solution

Reply:
I change a little bit the code to:

```
lgth <- length(args1)
  if (lgth > 0) {
    body_f <- paste0(
      '\told_attr <- purrr::keep(attributes(', args1[1],
      '), names(attributes(', args1[1], ')) %in% ',
      'c("dic_r2g", "dic_g2r", "wkdir", "experiment", "model", "class")) \n ',
      '\t', args1[1], ' <- as.data.frame(', args1[1], ') \n')
    if (lgth > 1) {
      f <- function(x) {
        paste0(
          '\told_attr <- purrr::keep(attributes(', x,
          '), names(attributes(', x, ')) %in% ',
          'c("dic_r2g", "dic_g2r", "wkdir", "experiment", "model", "class")) \n ',
          '\told_attr <- purrr::keep(old_attr, duplicated(old_attr) == FALSE)\n',
          '\t', x, ' <- as.data.frame(', x, ') \n')
      }
      body_f <- paste0(c(body_f, lapply(args1[-1], f)), collapse = "")
    }
  }
```
So, if the length of args1 > 1, the old_attr will contains both attributes so we need to discard the duplicate lines to avoid bug later

Reply:
Ah yes, indeed!